### PR TITLE
Reproducible Builds: trim embedded cgo paths

### PR DESCRIPTION
**Pull request by kpcyrd**
The Arch Linux reproducible builds setup currently fails to reproduce podman ([diff](https://reproducible.archlinux.org/api/v0/builds/147183/diffoscope)). The relevant part in the diff is:

```
│ │ ├── strings --all --bytes=8 {}
│ │ │ @@ -1,9 +1,9 @@
│ │ │  /lib64/ld-linux-x86-64.so.2
│ │ │ -fDMoJcLZqojBJnlXadN9/no_IKXu28dGv-5kgjWUm/HSPy4nrzjaE7jAmxaDUT/06FhXFIFJ4FwF81sy8x_
│ │ │ +CIrZsC53vVGulPBnzvSY/P_VErASmHt8np5NCPDQ8/LcpPc2AxUJaJ-MAKOAvh/xHYbWCnqxFpRwLK2eL7s
│ │ │  -*m4ibT|
│ │ │  _ITM_deregisterTMCloneTable
│ │ │  _ITM_registerTMCloneTable
│ │ │  pthread_mutexattr_setrobust
│ │ │  pthread_mutex_consistent
│ │ │  pthread_mutex_init
│ │ │  pthread_cond_wait
│ │ │ @@ -78810,15 +78810,15 @@
│ │ │  go/src/net/sockopt_posix.go
│ │ │  go/src/net/splice_linux.go
│ │ │  go/src/net/tcpsock_posix.go
│ │ │  go/src/net/tcpsockopt_posix.go
│ │ │  go/src/net/tcpsockopt_unix.go
│ │ │  go/src/net/udpsock_posix.go
│ │ │  go/src/net/unixsock_posix.go
│ │ │ -/tmp/go-build3474806368/b051/_cgo_gotypes.go
│ │ │ +/tmp/go-build3959924547/b051/_cgo_gotypes.go
│ │ │  go/src/net/cgo_resnew.go
│ │ │  go/src/net/cgo_unix.go
│ │ │  go/src/net/cgo_socknew.go
│ │ │  go/src/net/hook.go
│ │ │  go/src/net/hook_unix.go
│ │ │  go/src/net/url/url.go
│ │ │  go/src/crypto/dsa/dsa.go
│ │ │ @@ -78961,15 +78961,15 @@
│ │ │  go/src/strconv/atob.go
│ │ │  vendor/github.com/BurntSushi/toml/parse.go
│ │ │  vendor/github.com/BurntSushi/toml/type_check.go
│ │ │  vendor/github.com/BurntSushi/toml/type_fields.go
│ │ │  vendor/github.com/containers/image/v5/internal/rootless/rootless.go
│ │ │  go/src/os/user/lookup.go
│ │ │  go/src/os/user/user.go
│ │ │ -/tmp/go-build3474806368/b159/_cgo_gotypes.go
│ │ │ +/tmp/go-build3959924547/b159/_cgo_gotypes.go
│ │ │  go/src/os/user/cgo_lookup_unix.go
│ │ │  vendor/github.com/syndtr/gocapability/capability/capability.go
│ │ │  vendor/github.com/syndtr/gocapability/capability/capability_linux.go
│ │ │  vendor/github.com/syndtr/gocapability/capability/enum.go
│ │ │  vendor/github.com/syndtr/gocapability/capability/enum_gen.go
│ │ │  vendor/github.com/syndtr/gocapability/capability/syscall_linux.go
│ │ │  vendor/github.com/docker/go-units/duration.go
[...]
```

To fix the cgo paths, `-trimpath` needs to be passed in $GOFLAGS instead of `-gcflags` and `-asmflags`. This automatically resolves the other issues shown in the diff too. For more info on -trimpath with cgo see https://github.com/golang/go/issues/47256.

Thanks!

cc: @Foxboron 

[NO TESTS NEEDED]

### CODE REVIEW

**Comment by rhatdan** [comment]
You need to sign your commits.

git commit -a --amend -s
git push --force

You should also add [NO TESTS NEEDED] to your PR to avoid the next check.

**Comment by kpcyrd** [comment]
Done. I'm not sure why the tests are failing but I've added `[NO TESTS NEEDED]`.

Thanks!

**Comment by rhatdan** [comment]
Your PR is still not signed.

**Comment by kpcyrd** [comment]
The commit was signed but I never added my pgp key to my github account.

I've added the key, can you check again?

**Comment by rhatdan** [comment]
The problem is the commit message needs to have a line like

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

**Comment by kpcyrd** [comment]
Done!

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL
@mheon PTAL

**Comment by mheon** [comment]
I'd like a review from @lsm5 before merge, but LGTM

**Review by saschagrunert — APPROVED** [review]
LGTM

**Review by lsm5 — APPROVED** [review]
LGTM

**Comment by kpcyrd** [comment]
Thanks!

**Pull request merged**
